server/channels/app/notification_push.go

```diff
@@ -138,6 +140,25 @@ func (a *App) sendPushNotificationToAllSessions(msg *model.PushNotification, use
 }
 
 func (a *App) sendPushNotification(notification *PostNotification, user *model.User, explicitMention, channelWideMention bool, replyToThreadType string) {
+	cancelled := false
+	a.ch.RunMultiHook(func(hooks plugin.Hooks) bool {
+		rejectionReason := hooks.NotificationWillBePushed(&model.PluginPushNotification{
+			Post:    notification.Post.ForPlugin(),
+			Channel: notification.Channel,
+			UserID:  user.Id,
+		})
+		if rejectionReason != "" {
+			mlog.Error("Notification rejected by plugin.", mlog.String("reason", rejectionReason))
```

Comment:
I don't think this should be logged as an error, especially given it's a perfectly valid use case. `INFO` level would be more appropriate in my mind.

But honestly we may as well just return a boolean to indicate whether we should ignore the notification. If there's an error to be logged it's the plugin that should log it or return it as such (`error` interface).

Reply:
Agreed with the boolean approach.

Reply:
@streamer45 @lieut-data Not against it, just wanted to bring up two issues:
1. It breaks the pattern of hooks rejecting by returning reasons (we don't have to log the reason as an error of course).
2. If we change to a bool, we should probably return a zero-value as the "do nothing, let it continue". Which means returning a `true` if cancelling the push notification, which is a little unintuitive. We could mitigate a bit by changing the name... `NotificationWillBePushedCancel`? or `CancelNotificationWillBePushed`? I'm not thrilled with those. 

So, maybe I am against it? :) I think we should keep the pattern of returning a rejection reason, but log it as an INFO. Thoughts?

Reply:
Some counter thoughts:

1. A named return + proper documentation should be enough to clarify what the return value means.
2. We are already breaking consistency by not allowing modification.
3. The reasons we are returning a string in most of the other methods is because those are synchronous requests and we are actually returning it to the user. This is different as it's totally async and server initiated.

That said, my main concern was with the error log so if we can change that I am happy :)

Reply:
Yep, that makes sense to me. Made the change, thanks!